WordPress/gutenberg
# Templates API: return post modified datetime in response

## What?
All these file changes are so that we can add `modified` to the template and template part objects in the rest response for WP_REST_Templates_Controller 😄 

For theme templates, it will add the modified value when we have a post id (which is once the template is first save after user change)

The `modified` property will be returned when a template is created, updated and fetched.

## Why?

So that we can display the modified date in the UI. See: https://github.com/WordPress/gutenberg/issues/49597

## How?
Grabbing the value of `post_modified` from the post object and adding it to the response object

## Testing Instructions

```shell
npm run test:unit:php -- --filter Gutenberg_REST_Templates_Controller_Test
```

In the site editor, open a theme template and check the API response. There should be no modified field (because the template post record hasn't been created in the database yet.
<img width="700" alt="Screenshot 2023-06-09 at 1 23 50 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/90b7dced-51ef-46ae-915b-d497b2a4e62b">


Now make a few changes to the open theme template, and save. This will create a database record. Now refresh the page and inspect the API response (filter by `template`)

<img width="700" alt="Screenshot 2023-06-09 at 1 22 11 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/d989e781-7732-4a2b-a210-f6aa09b5bfed">

When creating a new custom templates, WordPress adds a database record immediately, and therefore the `modified` property is available.
